processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java

```diff
@@ -90,4 +95,16 @@ public CompletionTime processOneToOne(String tenantId, Span item) throws RetryAt
         return null;
     }
 
+    private void setFault(CompletionTime completionTime, List<BinaryAnnotation> binaryAnnotations) {
+        List<HttpCodesUtil.HttpCode> errorCodes =
+                HttpCodesUtil.getClientOrServerErrors(HttpCodesUtil.getHttpStatusCodes(binaryAnnotations));
+
+        if (errorCodes.size() > 0) {
```

Comment:
I would like to leave it in this more generic form. It can be also refactored to get any type of binary annotations. We will see what is needed in the future.